sdk/azcore/CHANGELOG.md

```diff
@@ -1,18 +1,12 @@
 # Release History
 
-## 1.16.1 (Unreleased)
+## 1.17.0 (2025-01-10)
```

Comment:
This is on a Friday. What if we pushed this up to `2025-01-07` so we can update all the things this week?

Reply:
👍 I like it